### PR TITLE
Create recommender's config classes

### PR DESCRIPTION
**Pull request by k-tahiro**
- Create recommender config classes for each recommender.
- Add `from_config` classmethod to use this config class to initialize recommender.

### CODE REVIEW

File: irspack/recommenders/ials.py [hunk 1]
```diff
@@ -69,11 +69,28 @@ class IALSConfigScaling(enum.Enum):
...
+    n_components: int = 20
+    alpha: float = 0.0
+    reg: float = 1e-3
+    confidence_scaling: str = "none"
+    epsilon: float = 1.0
+    init_std: float = 0.1
+    use_cg: bool = True
+    max_cg_steps: int = 3
+    random_seed: int = 42
+    validate_epoch: int = 5
```
Comment (tohtsky) [inline irspack/recommenders/ials.py:82]: kwargs

-  `validate_epoch`
- `score_degradation_max`
-  `max_epoch`

are common parameters for iALS, BPRFM, MultVAE.
Could you create a base for this with name e.g., "BaseEarlyStoppingRecomemnderConfig"?

**Review by tohtsky — COMMENTED** [review]
Please check the comment regarding the IALS's configuration.
By the way, it would be nice to have a meta-class check for config fields vs recommender args. Do you think it is feasible?

File: irspack/recommenders/ials.py (hunk 1 above)
Comment (k-tahiro) [reply irspack/recommenders/ials.py:82]: Thank you for your review.

Sure. I will add it.

**Comment by tohtsky** [comment]
Merged! Thank you!

**Pull request merged**